### PR TITLE
Pa 591 upgrade net6

### PR DESCRIPTION
**Pull request by alireid**

### CODE REVIEW

**Review by Pauljgraham — COMMENTED** [review]
Looks very good, a couple of tiny comments. And can we discuss how we should run this type of service locally. 

I know it has a Simulator endpoint in the solution so we can send an Event to the service. To me it always feels wrong to be adding test code into our soln, and to be fair, that endpoint will never be reachable in PROD. 

Now, is it worthwhile adding in these two types of projects
 
 SFA.DAS.ApprenticeAccounts.Jobs.MockServer - To mock/fake the outer API (so the service will not require the Outer API to be running
 
 SFA.DAS.ApprenticeAccounts.TestMessagePublisher - To send a message to service bus.
  
We use this pattern in TP Jobs and it generally we like this pattern as it limits the need to spin up many different interdependent projects and keeps this quite service standalone (locally). The Fake Services, turn on all the required services by default, but can choose to connect to the real services when required. 
 
PS I updated the ConfiureConfig section so it runs locally

File: src/SFA.DAS.ApprenticeAccounts.Jobs/Infrastructure/AzureServiceBusTriggerFunction.cs [hunk 1]
```diff
@@ -15,11 +16,11 @@ internal class AzureServiceBusTriggerFunction
 
         [FunctionName("ApprenticeshipCommitmentsJobs")]
         public async Task Run(
-            [ServiceBusTrigger(queueName: EndpointName, Connection = "AzureWebJobsServiceBus")] Message message,
+            [ServiceBusTrigger(queueName: EndpointName, Connection = "AzureWebJobsServiceBus")] ServiceBusReceivedMessage message,
```
[the remark this answers is not in the document]
Comment (alireid) [reply src/SFA.DAS.ApprenticeAccounts.Jobs/Infrastructure/AzureServiceBusTriggerFunction.cs:19]: Thanks Paul, good spot - fixed that up in the latest forthcoming commits

**Comment by alireid** [comment]
> Looks very good, a couple of tiny comments. And can we discuss how we should run this type of service locally.
> 
> I know it has a Simulator endpoint in the solution so we can send an Event to the service. To me it always feels wrong to be adding test code into our soln, and to be fair, that endpoint will never be reachable in PROD.
> 
> Now, is it worthwhile adding in these two types of projects
> 
> SFA.DAS.ApprenticeAccounts.Jobs.MockServer - To mock/fake the outer API (so the service will not require the Outer API to be running
> 
> SFA.DAS.ApprenticeAccounts.TestMessagePublisher - To send a message to service bus.
> 
> We use this pattern in TP Jobs and it generally we like this pattern as it limits the need to spin up many different interdependent projects and keeps this quite service standalone (locally). The Fake Services, turn on all the required services by default, but can choose to connect to the real services when required.
> 
> PS I updated the ConfiureConfig section so it runs locally

@Pauljgraham Really helpful comment thanks Paul, I've since added in both extra projects and will be available in forthcoming commits.

File: src/SFA.DAS.ApprenticeAccounts.Jobs/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            builder.Services.AddSingleton<IApimClientConfiguration>(x => x.GetRequiredService<ApiOptions>());
+
+            InitialiseNServiceBus();
+
+            builder.UseNServiceBus((IConfiguration appConfiguration) =>
+            {
+                var configuration = ServiceBusEndpointFactory.CreateSingleQueueConfiguration(QueueNames.ApprenticeshipCommitmentsJobs, appConfiguration, useManagedIdentity);
+                configuration.AdvancedConfiguration.UseNewtonsoftJsonSerializer();
+                configuration.AdvancedConfiguration.UseMessageConventions();
+                configuration.AdvancedConfiguration.EnableInstallers();
```
[the remark this answers is not in the document]
Comment (alireid) [reply src/SFA.DAS.ApprenticeAccounts.Jobs/Startup.cs:41]: Thanks Paul - I've actually tried that, which configuration can do chaining, I couldn't get the AdvancedConfiguration collection to honor chaining

File: src/SFA.DAS.ApprenticeAccounts.Jobs/Infrastructure/EsfaConfigurationExtension.cs [hunk 3]
```diff
@@ -33,6 +37,14 @@ public static void AddApplicationOptions(this IServiceCollection services)
...
         }
 
+        public static void ConfigureLogging(this ILoggingBuilder logBuilder)
+        {
+            logBuilder.AddFilter(typeof(Startup).Namespace, LogLevel.Information);
+            var rootDirectory = Path.GetFullPath(Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!, ".."));
+            var files = Directory.GetFiles(rootDirectory, "nlog.config", SearchOption.AllDirectories)[0];
+            logBuilder.AddNLog(files);
+        }
+
```
[the remark this answers is not in the document]
Comment (alireid) [reply src/SFA.DAS.ApprenticeAccounts.Jobs/Infrastructure/EsfaConfigurationExtension.cs:52]: Initially I incorporated builder.services.AddLogging() but good SonarCloud rejected it - will look to find a better way. In the meantime I've nuked this.

[the remark this answers is not in the document]
Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticeAccounts.Jobs/Infrastructure/EsfaConfigurationExtension.cs]: Yes, we see Sonar reject all the AddLogging() functions, but we need to log data. So, we have being ignoring this failure. Might be worth talking to DevOps about the logging approach we should take when updating the projects. 

Nothing formally filtered down to me about only AppInsight logging....

File: src/SFA.DAS.ApprenticeAccounts.Jobs/Startup.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticeAccounts.Jobs/Startup.cs:42]: weird, let me try it

[the remark this answers is not in the document]
Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticeAccounts.Jobs/Startup.cs:42]: My bad, yes it's a different endpoint.

**Review by Pauljgraham — APPROVED** [review]

**Pull request merged**